internal/venice-consumer/src/main/java/com/linkedin/venice/pubsub/api/PubSubMessage.java

```diff
@@ -0,0 +1,28 @@
+package com.linkedin.venice.pubsub.api;
+
+public interface PubSubMessage<K, V, TP extends PubSubTopicPartition, O> {
```

Comment:
Can we decouple`PubSubMessage` from  `PubSubTopicPartition`?

Reply:
I think `PubSubTopicPartition` and `PubSubMessage` are the interfaces we provide for client, and we do not want to have kafka term of `TopicPartition`. Do you mean message should get rid of topic partition information?

Reply:
I meant rather than extending `PubSubTopicPartition` we should add topic partition info as another field inside `PubSubMessage` and make it a sort of stand-alone object.

If it doesn't sound right, please feel free the comment.  

Thanks.

Reply:
Here what we provide is interfaces for `PubSubMessage`, implementation of  `PubSubMessage` like `KafkaMessage` in this change has the field of `topicPartition` as you pointed.